### ISSUE TITLE
How about an ESP32 version

### ISSUE THREAD

**Issue by JimPos** [gmag11/FSBrowserNG#33]
Are there any plans for an ESP32 version?

**Comment by species5618** [comment]
the latest verion of ESPAsyncWebServer  support the ESP32 based proudcta
so it should be possible, i will investigate

**Comment by JimPos** [comment]
Let me know if there is anything I can do to help

**Comment by JimPos** [comment]
Accidently closed this.
Hope it can happen.

**Comment by species5618** [comment]
i have been working throught a few items, it seem the wifi API uses slightly different event handling names , 
esp32 library has difeerent time / ticker includes, 
may take a while

**Comment by JimPos** [comment]
I did get the ESPAsyncWebServer example working for ESP32 with only a few minor changes,
so hopefully it won't be too bad.

**Comment by dadosch** [comment]
Are there any news?

**Comment by gmag11** [comment]
Hello. No news from my side, I'm sorry.

**Comment by JimPos** [comment]
@dadosch After considerable effort I actually got a version to compile and run on an ESP32.
However it might not be worth it.
I forced a default wifi configuration to eliminate some compiler errors in the wifi sections.
The NTPClientLib used is not compatible with ESP32 so I commented out all NTP or time references.
The OTA update sections have many incompatibilities, so I removed OTA.
There are some signicant differences between ESP8266 SPIFFS and ESP32 SPIFFS so
   handleFileRead and handleFileList needed to be modified/rewritten.

At that point it could be compiled and run, but it was easier to start from scratch and 
just expand the  example in the ESPAsyncWebserver by  @me-no-dev.

**Comment by gmag11** [comment]
For your information. I'm working on new version of NTPClientLib. On https://github.com/gmag11/NtpClient/tree/pre2.5.0 you can find latest development. Now It is compatible with ESP8266, MKR1000 and ESP32.

**Comment by gmag11** [comment]
@JimPos I agree. I developed this quite long ago, and finally never used it. I learned a lot about C++ in Arduino and ESP8266 but the code is too big. I'd like to turn this into a simpler web server, that publishes what is inside SPIFFS. But currently I occupied with other projects :-)

**Comment by JimPos** [comment]
@gmag11 the new version of your NtpClientLib  https://github.com/gmag11/NtpClient/tree/pre2.5.0  seems to work fine.
There is also now an ESP32 version of Ticker [https://github.com/bertmelis/Ticker-esp32] .
Using these allowed me to put back a lot of lines I had commented out.
WiFi events are still a problem, so I'm currently hard coding my ssid and password.
The "AP Button" code thinks I'm always pressing the button, so that's commented out too.
Many other problems with FileSystem/SPIFFS were solved by globally replacing '_fs->' with 'SPIFFS.'
OTA is still commented out.

Like you, I used your code to better understand how the Async Web server and FS Browser worked, 
But I find it a little too large and non-generic as a library.
